client/CPlayerInterface.cpp

```diff
@@ -1685,16 +1686,39 @@ void CPlayerInterface::showShipyardDialogOrProblemPopup(const IShipyard *obj)
 
 void CPlayerInterface::requestReturningToMainMenu(bool won)
 {
+	HighScoreParameter param;

```

Comment:
Perhaps compute this structure on server and send it as part of netpack?

Reply:
I don't want do get in to deep in server/client code. :)